google-cloud-storage/src/main/java/com/google/cloud/storage/multipartupload/model/CreateMultipartUploadRequest.java

```diff
@@ -0,0 +1,284 @@
+  private final ObjectLockMode objectLockMode;
+  private final Date objectLockRetainUntilDate;
+
+  private CreateMultipartUploadRequest(Builder builder) {
+    this.bucket = builder.bucket;
+    this.key = builder.key;
+    this.cannedAcl = builder.cannedAcl;
+    this.contentType = builder.contentType;
+    this.metadata = builder.metadata;
+    this.storageClass = builder.storageClass;
+    this.customTime = builder.customTime;
+    this.kmsKeyName = builder.kmsKeyName;
+    this.objectLockMode = builder.objectLockMode;
+    this.objectLockRetainUntilDate = builder.objectLockRetainUntilDate;
+  }
+
+  public String bucket() {
+    return bucket;
+  }
+
+  public String key() {
+    return key;
+  }
+
+  public PredefinedAcl getCannedAcl() {
+    return cannedAcl;
+  }
+
+  public String getContentType() {
+    return contentType;
+  }
+
+  public Map<String, String> getMetadata() {
+    return metadata;
+  }
+
+  public StorageClass getStorageClass() {
+    return storageClass;
+  }
+
+  public Date getCustomTime() {
+    return customTime;
+  }
+
+  public String getKmsKeyName() {
+    return kmsKeyName;
+  }
+
+  public ObjectLockMode getObjectLockMode() {
+    return objectLockMode;
+  }
+
+  public Date getObjectLockRetainUntilDate() {
+    return objectLockRetainUntilDate;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof CreateMultipartUploadRequest)) {
+      return false;
+    }
+    CreateMultipartUploadRequest that = (CreateMultipartUploadRequest) o;
+    return Objects.equals(bucket, that.bucket)
+        && Objects.equals(key, that.key)
+        && cannedAcl == that.cannedAcl
+        && Objects.equals(contentType, that.contentType)
+        && Objects.equals(metadata, that.metadata)
+        && Objects.equals(storageClass, that.storageClass)
+        && Objects.equals(customTime, that.customTime)
+        && Objects.equals(kmsKeyName, that.kmsKeyName)
+        && objectLockMode == that.objectLockMode
+        && Objects.equals(objectLockRetainUntilDate, that.objectLockRetainUntilDate);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(
+        bucket,
+        key,
+        cannedAcl,
+        contentType,
+        metadata,
+        storageClass,
+        customTime,
+        kmsKeyName,
+        objectLockMode,
+        objectLockRetainUntilDate);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("bucket", bucket)
+        .add("key", key)
+        .add("cannedAcl", cannedAcl)
+        .add("contentType", contentType)
+        .add("metadata", metadata)
+        .add("storageClass", storageClass)
+        .add("customTime", customTime)
+        .add("kmsKeyName", kmsKeyName)
+        .add("objectLockMode", objectLockMode)
+        .add("objectLockRetainUntilDate", objectLockRetainUntilDate)
+        .toString();
+  }
+
+  public static Builder builder() {
```

Comment:
nit: style consistency


```suggestion
  public static Builder newBuilder() {
```